appveyor-ubuntu.yml

```diff
@@ -338,6 +339,17 @@ for:
       - "pip install -e \".[dev]\""
       - sh: "pytest -vv tests/integration/local --json-report --json-report-file=TEST_REPORT-integration-local.json"
 
+  # End-to-end testing
+  -
+    matrix:
+      only:
+        - image: Ubuntu
+          configuration: EndToEndTesting
+
+    test_script:
+      - "pip install -e \".[dev]\""
+      - sh: "pytest -vv tests/end_to_end --json-report --json-report-file=TEST_REPORT-end-to-end.json"
```

Comment:
Any chance we can run them in parallel (with providing `--dist loadscope`, which will queue tests from same class into same thread)?

Reply:
Does it require any additional configuration other than providing that one flag?

Reply:
We will need to replace the line above with something like;
```
pytest -vv -n N --reruns M --dist loadscope tests/end_to_end --json-report --json-report-file=TEST_REPORT-end-to-end.json
```
`N` is number of threads running in parallel, and `M` will be number of retries that we need to do for each test case in case they fail. From their documentation (https://pytest-xdist.readthedocs.io/en/stable/distribution.html);
> `--dist loadscope`: Tests are grouped by module for test functions and by class for test methods. Groups are distributed to available workers as whole units. This guarantees that all tests in a group run in the same process. This can be useful if you have expensive module-level or class-level fixtures. Grouping by class takes priority over grouping by module.

Reply:
This took some tinkering with the init stage since the tests write to a shared resource. I was able to get it to run in parallel by locking the `~/.aws-sam/` directory using `Filelock`.